### ISSUE TITLE
Duplicate Session Cookies

### ISSUE THREAD

**Issue by caffeineinc** [silverstripe/silverstripe-framework#8543]
## Affected Version

4.x.x

## Description

SilverStripe 4 changes how php sessions are bootstrapped, but with the current configuration the session related cookies are duplicated, re-sent on every request. This causes issues with WAF's and cookie persistence as there is always a "Set-Cookie" header. 

Session should only be bootstrapped when/if there is modification (writes or changes) to the session data to be persisted. If there's a form on a page, this will create a session (to save the CSRF token) - The current behaviour creates the session,  but writes two cookies as [session_start()](https://github.com/silverstripe/silverstripe-framework/blob/4.0/src/Control/Session.php#L278) starts a cookie, and outputs a header. Then the [timeout](https://github.com/silverstripe/silverstripe-framework/blob/4.0/src/Control/Session.php#L288) code resets the header values. 

```
< Set-Cookie: PHPSESSID=<sessionid>; expires=Mon, 29-Oct-2018 21:38:08 GMT; Max-Age=3600; path=/; HttpOnly
< Set-Cookie: PHPSESSID=<sessionid>; expires=Mon, 29-Oct-2018 21:38:08 GMT; Max-Age=3600; path=/; HttpOnly
```

The problem here's is the second header is unneeded, but  also is being output even if we have a session bootstrapped elsewhere. This means you **always** get a Set-Cookie on every requests where session is used. this breaks WAFs that are trying to use persistence to direct sessions to the same origins when load-balancing.  

## Steps to Reproduce

- `SilverStripe\Control\Session` config parameter `timeout` must be `!= 0`
- Create a page in SilverStripe with a form with a CSRF token (default is enabled)
- Visit the page and note the headers with curl. `curl -svok /dev/null https://mysite.com/my-form-page`
- Note, duplicate headers in response
- Visit the page again with the headers and cookie `curl -svok --cookie "PHPSESSID=<PHPSESSID from-request-above>" /dev/null https://mysite.com/my-form-page` 
- Note `Set-Cookie` header with the same PHPSESSID value as your cookie. 

The second request shouldn't return another `Set-Cookie` unless the session has been destroyed, updated, or is no longer valid.

Related to https://github.com/silverstripe/silverstripe-hybridsessions/issues/41

# Pull Request

 * [x] https://github.com/silverstripe/silverstripe-framework/pull/8544
 * [x] https://github.com/silverstripe/silverstripe-framework/pull/8639

**Comment by chillu** [comment]
Talked to @caffeineinc, they're fine with delaying this until 4.3.1, since the cutoff for 4.3.0 has already happened. There's a very limited number of customers known to be affected by this, which we can contact directly in the meantime.

**Comment by chillu** [comment]
@caffeineinc I'm assuming you're happy with this in the next CWP release (Mar 2019), and in a core release soon (4.3.1) which you can point anyone with problems to?

**Comment by dnsl48** [comment]
Merged to `4.3`, will be in the `4.3.1` release.
